### PR TITLE
decoupling ESP3 parsing from ESP3 Typed parsing

### PR DESCRIPTION
**Pull request by Holger-Will**
have a second parser that inherits from simple ESP3Parser and which is able to parse ESP3Packets by type.

### CODE REVIEW

**Comment by Holger-Will** [comment]
this way we can export multible parsers for all needs.
the second parser could live in a second repository though...

**Comment by RafaelKa** [comment]
can you rebase, please

**Comment by Holger-Will** [comment]
this is what 1.0.0 could look like. We have a test coverage of 100%, packet detection and packet type parsing is properly decoupled, all ERP3 Packet types are implemented, and we have standard coding style in place. I think the naming convention is readable and consistent.
The only thing that is not yet in this branch is logging.

**Comment by Holger-Will** [comment]
fix #14 #17 #18 #19 #20

File: ESP3Packet.js [hunk 1]
```diff
@@ -1,44 +1,5 @@
...
- * |               |
- * +---------------+ - + -
- * |   CRC8 Data   |   1
- * +---------------+ -----
- *
- * also full ESP3 Packet Length = 7 + X(Data length) + Y(Optional Data length)
- */
-
 class ESP3Packet {
   constructor () {
```
Comment (RafaelKa) [inline ESP3Packet.js]: We can save some lines on concrete packets by using the construct from #36 

https://github.com/RafaelKa/node-serialport-enocean-parser/pull/36/files#diff-c6ea59fd55a6999b6c8b3838f6146175R8

 see ESP3/Packet.js and PacketTypes/RADIO_ERP1.js from the branch

**Review by RafaelKa — COMMENTED** [review]
Wow,
I think we can merge, after small fixes in constructors of Packets see comment and example in #36
And I think we should lock the packets before emitting.

:+1:

**Comment by Holger-Will** [comment]
can we land this, and do optimizations and improvements in separat PRs?

File: ESP3Packet.js (hunk 1 above)
Comment (Holger-Will) [reply ESP3Packet.js]: can we land this, and do optimizations and improvements in separat PRs?

Comment (Holger-Will) [reply ESP3Packet.js]: and please don't freeze the objects. there is no benefit in doing so, and in turn takes away some flexibility for users of this module...

**Pull request merged**